Asgard.Console/MessageHistory.cs

```diff
@@ -48,12 +49,15 @@ public MessageHistory(ICbusMessenger cbusMessenger)
 
             cbusMessenger.MessageReceived += (sender, e) =>
```

Comment:
I wonder if the messenger should now be raising multiple events on message received - as well as keeping this one, also introduce a StandardMessageReceived and an ExtendedMessageReceived that has the appropriate things already on it

Reply:
Having come across multiple places in the core code that are now having to check for `is CbusStandardMessage` before continuing - think it definitely makes sense to split that out at a higher level, so the places that only care about standard messages don't need to all individually check.  Think there's a fine line that maybe we're missing somewhere yet between a core CBUS message and the extended ones that won't be used the bulk of the time - perhaps there's another split in layers that needs to happen somewhere to make it make sense fully.

Reply:
I'm kind of wanting to keep the layers more or less aligned with the ISO model. I think GridConnect would be the Network layer; and CBUS the Transport layer. That would make the *Manager classes the session layer. Raising multiple events might well be the way to not need to add extra layers, but allow higher layers only have to deal with what they want to.